src/todo_collection.js

```diff
@@ -0,0 +1,186 @@
+              item.classList.remove('completed');
+            }
+          });
+
+          item.addEventListener('click', () => {
+            const inputList = Array.from(document.querySelectorAll('.editInput'));
+            const editText = Array.from(document.querySelectorAll('.editText'));
+            const bgInput = Array.from(document.querySelectorAll('.main__item'));
+            const svgBtn = Array.from(document.querySelectorAll('.svg'));
+            if (editInput.classList.contains('hidden')) {
+              const valInput = inputList.find((item) => !item.classList.contains('hidden'));
+              const valText = editText.find((item) => item.classList.contains('hidden'));
+              const valBg = bgInput.find((item) => item.classList.contains('bg-yellow'));
+              const valSvg = svgBtn.find((item) => item.classList.contains('trash'));
+              const svgList = document.querySelectorAll('.svg');
+
+              if (valInput) {
+                valInput.classList.add('hidden');
+                valText.classList.remove('hidden');
+                valBg.classList.remove('bg-yellow');
+                valSvg.classList.remove('trash');
+              }
+
+              editInput.classList.remove('hidden');
+              editInput.focus();
+              itemText.classList.add('hidden');
+              task.classList.add('bg-yellow');
+              svg.classList.add('trash');
+
+              const svgTrash = document.querySelector('.trash');
+
+              svgTrash.addEventListener('click', () => {
+                this.todoData.splice(i, 1);
+                for (let j = i; j < this.todoData.length; j += 1) this.todoData[j].index = j;
+                localStorage.setItem('todoData', JSON.stringify(this.todoData));
+                this.render();
+              });
+
+              svgList.forEach((item) => {
+                if (item.classList.contains('trash')) {
+                  item.innerHTML = `
+                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-trash" width="24" height="24" viewBox="0 0 24 24" stroke-width="1.5" stroke="#cccccc" fill="none" stroke-linecap="round" stroke-linejoin="round">
+                      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
+                      <path d="M4 7l16 0" />
+                      <path d="M10 11l0 6" />
+                      <path d="M14 11l0 6" />
+                      <path d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12" />
+                      <path d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3" />
+                    </svg>
+                  `;
+                } else {
+                  item.innerHTML = `
+                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-dots-vertical" width="24" height="24" viewBox="0 0 24 24" stroke-width="1.5" stroke="#cccccc" fill="none" stroke-linecap="round" stroke-linejoin="round">
```

Comment:
Please try to avoid repeating yourself try to reuse this segment by adding it to a variable, this will help you in creating a  neat and DRY! code

Reply:
Solved